src/components/Trade/TradeConfirm/AssetToAsset.tsx

```diff
@@ -21,42 +22,42 @@ export const AssetToAsset: FC<AssetToAssetProps> = ({
   sellColor = '#2775CA',
   buyColor = '#2775CA',
   status,
+  isSubmitting,
   ...rest
 }) => {
   const gray = useColorModeValue('white', 'gray.750')
   const red = useColorModeValue('white', 'red.500')
   const green = useColorModeValue('white', 'green.500')
 
-  const renderIcon = () => {
-    switch (status) {
-      case TxStatus.Confirmed:
+  const icon = useMemo(() => {
+    switch (true) {
+      case status === TxStatus.Confirmed:
         return (
           <Circle bg={green} size='100%'>
             <CheckIcon />
           </Circle>
         )
-      case TxStatus.Failed:
+      case status === TxStatus.Failed:
         return (
           <Circle bg={red} size='100%'>
             <CloseIcon p={1} />
           </Circle>
         )
-      case TxStatus.Pending:
+      // when the trade is submitting, treat unknown status as pending so the spinner spins
+      case status === TxStatus.Pending || (isSubmitting && status === TxStatus.Unknown):
```

Comment:
not a user facing change, but allows the new UI to show the spinner while a trade is executing for unknown status. This prevents cowswap from sitting there doing nothing without a spinner and suddenly jumping to green tick without a spinner - in other words prevents user thinking app is broken when they sign the tx.